src/utils.ts

```diff
@@ -227,39 +145,43 @@ export function getTSConfig(globals, collectCoverage: boolean = false) {
   // see https://github.com/kulshekhar/ts-jest/issues/309
   delete config.outDir;
 
-  // Note: If we had to read the inline configuration, it's required to set the fields
-  // to their string properties, and convert the result accordingly afterwards.
-  // In case of an external file, reading the config file already converted it as well, and
-  // an additional attempt would lead to errors.
-  let result;
-  if (isReferencedExternalFile) {
-    config.jsx = config.jsx || tsc.JsxEmit.React;
-    config.module = config.module || tsc.ModuleKind.CommonJS;
-    if (config.allowSyntheticDefaultImports && !skipBabel) {
-      // compile ts to es2015 and transform with babel afterwards
-      config.module = tsc.ModuleKind.ES2015;
-    }
-    result = config;
-  } else {
-    config.jsx = config.jsx || 'react';
-    config.module = config.module || 'commonjs';
-    if (config.allowSyntheticDefaultImports && !skipBabel) {
-      // compile ts to es2015 and transform with babel afterwards
-      config.module = 'es2015';
-    }
-    const converted = tsc.convertCompilerOptionsFromJson(config, undefined);
-    if (converted.errors && converted.errors.length > 0) {
-      const formattedErrors = formatTscParserErrors(converted.errors);
-      throw new Error(
-        `Some errors occurred while attempting to convert ${JSON.stringify(
-          config,
-        )}: ${formattedErrors}`,
-      );
-    }
-    result = converted.options;
+  if (configPath === path.join(getStartDir(), 'tsconfig.json')) {
```

Comment:
Is this to ensure that if the file is just called tsconfig.json we overwrite the module, and if it's called something like tsconfig.test.json we expect people to know what they're doing? I like the idea, but I feel like it could lead to trouble.

Reply:
The thought process was that because most people don't create a custom tsconfig for testing, it's safe to override the `module` to `commonjs` during testing. But if they do have a custom tsconfig, they're more likely to have thought through what they want and any module change they made should be used